### PR TITLE
[grafana] Add support for envValueFrom in datasources container and fix typo in dashboards container config

### LINKED ISSUE #3186: [grafana] add support for envValueFrom in datasource init and side car container and fix typo in dashboards container
Add support for `envValueFrom`  to set Env vars from ConfigMaps or Secrets for both init datasources container and datasources sidecar container.
Also fix dashboards container config which is currently referring to datasources values.

```
      {{- range $key, $value := .Values.sidecar.datasources.envValueFrom }}
      - name: {{ $key | quote }}
        valueFrom:
          {{- tpl (toYaml $value) $ | nindent 10 }}
```

### PR DESCRIPTION
**Pull request by rgaduput**
Fixes #3186

### CODE REVIEW

**Review by zanhsieh — APPROVED** [review]

**Review by jkroepke — CHANGES_REQUESTED** [review]
Hi,

to increase our test coverage,  could you please fillout some new extra values in the CI test config that this values are covered by CI?

https://github.com/grafana/helm-charts/blob/main/charts/grafana/ci/with-extraconfigmapmounts-values.yaml

For extraVolumes / extraVolumeMounts, empty would be sufficient. 

For extraConfigmapMounts / extraSecretMounts, you may have to setup own ConfigMaps / Secrets, which can be done via extraObject

**Comment by rgaduput** [comment]
> Hi,
> 
> to increase our test coverage, could you please fillout some new extra values in the CI test config that this values are covered by CI?
> 
> https://github.com/grafana/helm-charts/blob/main/charts/grafana/ci/with-extraconfigmapmounts-values.yaml
> 
> For extraVolumes / extraVolumeMounts, empty would be sufficient.
> 
> For extraConfigmapMounts / extraSecretMounts, you may have to setup own ConfigMaps / Secrets, which can be done via extraObject

@jkroepke tried to add some tests to cover the changes I am introducing, Could you please take a look.

File: charts/grafana/ci/with-sidecars-envvaluefrom-values.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      name: '{{ include "grafana.fullname" . }}'
+    data:
+      var1: "value1"
+  - apiVersion: v1
+    kind: Secret
+    metadata:
+      name: '{{ include "grafana.fullname" . }}'
+    type: Opaque
+    data:
+      var2: "value2"
```
Comment (jkroepke) [inline charts/grafana/ci/with-sidecars-envvaluefrom-values.yaml:14]: data in secrets must be base64 encoded

```suggestion
      var2: "dmFsdWUy"
```

File: charts/grafana/ci/with-sidecars-envvaluefrom-values.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: '{{ include "grafana.fullname" . }}'
+    data:
+      var1: "value1"
+  - apiVersion: v1
+    kind: Secret
+    metadata:
+      name: '{{ include "grafana.fullname" . }}'
```
Comment (jkroepke) [inline charts/grafana/ci/with-sidecars-envvaluefrom-values.yaml:11]: CI says a secret with that name already exists.

Could you add a suffix here like test? please update the references in envValueFrom

Comment (rgaduput) [reply charts/grafana/ci/with-sidecars-envvaluefrom-values.yaml]: @jkroepke done.

File: charts/grafana/values.yaml [hunk 3]
```diff
@@ -929,6 +929,19 @@ sidecar:
...
+    ## "valueFrom" environment variable references that will be added to deployment pods. Name is templated.
+    ## ref: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#envvarsource-v1-core
+    ## Renders in container spec as:
+    ##   env:
+    ##     ...
+    ##     - name: <key>
+    ##       valueFrom:
+    ##         <value rendered as YAML>
+    envValueFrom: {}
+      #  ENV_NAME:
```
Comment (jkroepke) [inline charts/grafana/values.yaml:941]:
```suggestion
   #  ENV_NAME:
```

File: charts/grafana/values.yaml [hunk 4]
```diff
@@ -994,7 +1007,19 @@ sidecar:
...
+    ## ref: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#envvarsource-v1-core
+    ## Renders in container spec as:
+    ##   env:
+    ##     ...
+    ##     - name: <key>
+    ##       valueFrom:
+    ##         <value rendered as YAML>
     envValueFrom: {}
+      #  ENV_NAME:
+      #    configMapKeyRef:
```
Comment (jkroepke) [inline charts/grafana/values.yaml:1020]:
```suggestion
    #  ENV_NAME:
    #    configMapKeyRef:
```

File: charts/grafana/values.yaml [hunk 5]
```diff
@@ -929,6 +929,19 @@ sidecar:
...
+    ## "valueFrom" environment variable references that will be added to deployment pods. Name is templated.
+    ## ref: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#envvarsource-v1-core
+    ## Renders in container spec as:
+    ##   env:
+    ##     ...
+    ##     - name: <key>
+    ##       valueFrom:
+    ##         <value rendered as YAML>
+    envValueFrom: {}
+   #  ENV_NAME:
```
Comment (jkroepke) [inline charts/grafana/values.yaml:941]:
```suggestion
    #  ENV_NAME:
```

**Review by jkroepke — APPROVED** [review]
Thanks!

**Pull request merged**